src/main/scala/strawman/collection/IterableFactories.scala

```diff
@@ -4,23 +4,61 @@ package collection
 import strawman.collection.mutable.Builder
 
 import scala.Int
+import scala.annotation.unchecked.uncheckedVariance
+
+trait FromIterableBase[-B] {
+  type To[X]
+  def fromIterable[E <: B](it: Iterable[E]): To[E]
+}
 
 /** Base trait for instances that can construct a collection from an iterable */
-trait FromIterable[+C[X] <: Iterable[X]] {
-  def fromIterable[B](it: Iterable[B]): C[B]
+trait FromIterable[-B, +C[_]] extends FromIterableBase[B] {
```

Comment:
Why do we need the bound here? It would be great if we could avoid the additonal `B` parameter.